src/common/data_chunk/data_chunk_collection.cpp

```diff
@@ -44,20 +44,20 @@ void DataChunkCollection::initTypes(DataChunk& chunk) {
     types.clear();
     types.reserve(chunk.getNumValueVectors());
     for (auto vectorIdx = 0u; vectorIdx < chunk.getNumValueVectors(); vectorIdx++) {
-        types.push_back(chunk.getValueVector(vectorIdx)->dataType);
+        types.push_back(chunk.getValueVector(vectorIdx)->dataType.copy());
     }
 }
 
 void DataChunkCollection::allocateChunk(DataChunk& chunk) {
     if (chunks.empty()) {
         types.reserve(chunk.getNumValueVectors());
         for (auto vectorIdx = 0u; vectorIdx < chunk.getNumValueVectors(); vectorIdx++) {
-            types.push_back(chunk.getValueVector(vectorIdx)->dataType);
+            types.push_back(chunk.getValueVector(vectorIdx)->dataType.copy());
```

Comment:
@ray6080 should check whether this runtime copy is necessary or not.

Reply:
I think so. Otherwise we should consider getting rid of `types` as a field.